tests/components/alexa/test_capabilities.py

```diff
@@ -243,6 +278,102 @@ async def test_api_select_input(
     assert call.data["source"] == source_list[idx]
 
 
+@pytest.mark.parametrize(
+    (
+        "target_activity",
+        "activity_list",
+        "current_activity_index",
+        "target_activity_index",
+    ),
+    [
+        ("TV", ["TV", "MUSIC", "DVD"], 1, 0),
+        ("MUSIC", ["TV", "MUSIC", "DVD", 1000], 0, 1),
+        ("DVD", ["TV", "MUSIC", "DVD", None], 0, 2),
+        ("BAD DEVICE", ["TV", "MUSIC", "DVD"], 0, None),
+        ("TV", ["TV"], 0, 0),
+        ("BAD DEVICE", [], None, None),
+    ],
+)
+async def test_api_select_activity(
+    hass: HomeAssistant,
+    target_activity: str,
+    activity_list: list[str],
+    current_activity_index: int | None,
+    target_activity_index: int | None,
+) -> None:
+    """Test api set activity process."""
+    curr_activty = (
+        activity_list[current_activity_index] if current_activity_index else "None"
+    )
+    hass.states.async_set(
+        "remote.test",
+        "off",
+        {
+            "current_activity": curr_activty,
+            "activity_list": activity_list,
+        },
+    )
+    # test where no source matches
+    if target_activity_index is None:
```

Comment:
This case should be it's own separate test since we're not using the whole test body anyway in this case.

Reply:
Agree: https://github.com/home-assistant/core/pull/121627